--- crash-trace-command-1.0/Makefile.orig
+++ crash-trace-command-1.0/Makefile
@@ -28,4 +28,4 @@ INCDIR=/usr/include/crash
 all: trace.so
 	
 trace.so: $(INCDIR)/defs.h trace.c 
-	gcc -Wall -I$(INCDIR) -nostartfiles -shared -rdynamic -o trace.so trace.c -fPIC -D$(TARGET) $(TARGET_CFLAGS)
+	gcc $(RPM_OPT_FLAGS) -Wall -I$(INCDIR) -nostartfiles -shared -rdynamic -o trace.so trace.c -fPIC -D$(TARGET) $(TARGET_CFLAGS)
--- crash-trace-command-1.0/trace.c.orig
+++ crash-trace-command-1.0/trace.c
@@ -473,6 +473,7 @@ static void ftrace_destroy(void)
 static int ftrace_dump_page(FILE *out, ulong page, void *page_tmp)
 {
 	ulong raw_page;
+	size_t bytes_written __attribute__ ((__unused__));
 
 	read_value(raw_page, page, buffer_page, page);
 
@@ -480,7 +481,7 @@ static int ftrace_dump_page(FILE *out, u
 			RETURN_ON_ERROR))
 		goto out_fail;
 
-	fwrite(page_tmp, 1, PAGESIZE(), out);
+	bytes_written = fwrite(page_tmp, 1, PAGESIZE(), out);
 
 	return 0;
 
@@ -1180,7 +1181,7 @@ static int ring_buffer_stream_pop_event(
 		struct ftrace_event *res)
 {
 	int cpu, min_cpu = -1;
-	u64 ts, min_ts;
+	u64 ts, min_ts = 0;
 
 	res->data = NULL;
 
